File: app/src/main/java/ru/ulstu_team/ulstuschedule/data/model/LessonComparator.java
package ru.ulstu_team.ulstuschedule.data.model;

import java.util.Comparator;

public class LessonComparator implements Comparator<Lesson> {
    @Override
    public int compare(Lesson lhs, Lesson rhs) {
        return getRate(rhs) - getRate(lhs);
    }

    private int getRate(Lesson lesson) {
        int rate = 0;
        rate += (3 - lesson.getNumberOfWeek()) * 100;
        rate += (7 - lesson.getDayOfWeek()) * 10;
        rate += (9 - lesson.getNumber());
        return rate;
    }
}

File: app/src/main/java/ru/ulstu_team/ulstuschedule/data/model/LessonComparator.kt
package ru.ulstu_team.ulstuschedule.data.model

import java.util.Comparator

class LessonComparator : Comparator<Lesson> {
    override fun compare(lhs: Lesson, rhs: Lesson): Int {
        return getRate(rhs) - getRate(lhs)
    }

    private fun getRate(lesson: Lesson): Int {
        var rate = 0
        rate += (3 - lesson.numberOfWeek) * 100
        rate += (7 - lesson.dayOfWeek) * 10
        rate += 9 - lesson.number
        return rate
    }
}
